src/dmd/cparse.d

```diff
@@ -2758,19 +2759,29 @@ final class CParser(AST) : Parser!AST
         {
             tag = token.ident;
             nextToken();
+            if (token.value == TOK.semicolon)
+            {
+                /* `struct tag;` always results in a declaration
```

Comment:
`struct-or-union tag`?

Reply:
The comment cases for struct and union are identical, so I just use `struct` for brevity.

Reply:
I meant the code also just generates a struct declaration. Shouldn't structOrUnion also be tested here as well?

Reply:
yes, good catch